can't build engine in mac os x

mac version 10.14.3

error info:
```
# robert @ Paul in /opt/openssl-1.1.1/engine/build on git:master o [14:01:21]
$ sudo cmake --build . --target install --config Release
Scanning dependencies of target test_tlstree
[  2%] Building C object CMakeFiles/test_tlstree.dir/test_tlstree.c.o
[  4%] Linking C executable bin/test_tlstree
[  4%] Built target test_tlstree
Scanning dependencies of target gost_core
[  6%] Building C object CMakeFiles/gost_core.dir/gost89.c.o
[  8%] Building C object CMakeFiles/gost_core.dir/gosthash.c.o
[ 10%] Building C object CMakeFiles/gost_core.dir/gosthash2012.c.o
[ 12%] Building C object CMakeFiles/gost_core.dir/gost_grasshopper_core.c.o
[ 14%] Building C object CMakeFiles/gost_core.dir/gost_grasshopper_defines.c.o
[ 16%] Building C object CMakeFiles/gost_core.dir/gost_grasshopper_galois_precompiled.c.o
[ 18%] Building C object CMakeFiles/gost_core.dir/gost_grasshopper_precompiled.c.o
[ 20%] Building C object CMakeFiles/gost_core.dir/gost_grasshopper_cipher.c.o
[ 22%] Building C object CMakeFiles/gost_core.dir/gost_ec_keyx.c.o
[ 24%] Building C object CMakeFiles/gost_core.dir/gost_ec_sign.c.o
[ 26%] Building C object CMakeFiles/gost_core.dir/gost_omac.c.o
[ 28%] Building C object CMakeFiles/gost_core.dir/gost_omac_acpkm.c.o
[ 30%] Linking C static library libgost_core.a
[ 30%] Built target gost_core
Scanning dependencies of target gost12sum
[ 32%] Building C object CMakeFiles/gost12sum.dir/gost12sum.c.o
[ 34%] Linking C executable bin/gost12sum
[ 34%] Built target gost12sum
Scanning dependencies of target gost_engine
[ 36%] Building C object CMakeFiles/gost_engine.dir/e_gost_err.c.o
[ 38%] Building C object CMakeFiles/gost_engine.dir/gost_asn1.c.o
[ 40%] Building C object CMakeFiles/gost_engine.dir/gost_crypt.c.o
[ 42%] Building C object CMakeFiles/gost_engine.dir/gost_ctl.c.o
[ 44%] Building C object CMakeFiles/gost_engine.dir/gost_eng.c.o
[ 46%] Building C object CMakeFiles/gost_engine.dir/gost_keywrap.c.o
[ 48%] Building C object CMakeFiles/gost_engine.dir/gost_params.c.o
[ 50%] Building C object CMakeFiles/gost_engine.dir/gost_keyexpimp.c.o
[ 52%] Building C object CMakeFiles/gost_engine.dir/gost_ameth.c.o
[ 54%] Building C object CMakeFiles/gost_engine.dir/gost_md.c.o
[ 56%] Building C object CMakeFiles/gost_engine.dir/gost_md2012.c.o
[ 58%] Building C object CMakeFiles/gost_engine.dir/gost_pmeth.c.o
[ 60%] Building C object CMakeFiles/gost_engine.dir/gost_omac.c.o
[ 62%] Building C object CMakeFiles/gost_engine.dir/gost_omac_acpkm.c.o
[ 64%] Linking C shared module bin/gost.so
[ 64%] Built target gost_engine
Scanning dependencies of target gostsum
[ 66%] Building C object CMakeFiles/gostsum.dir/gostsum.c.o
[ 68%] Linking C executable bin/gostsum
[ 68%] Built target gostsum
Scanning dependencies of target test_params
[ 70%] Building C object CMakeFiles/test_params.dir/test_params.c.o
[ 72%] Linking C executable bin/test_params
ld: can't link with bundle (MH_BUNDLE) only dylibs (MH_DYLIB) file 'bin/gost.so' for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [bin/test_params] Error 1
make[1]: *** [CMakeFiles/test_params.dir/all] Error 2
make: *** [all] Error 2
```
can you help me?

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -26,6 +26,9 @@ endif()
 
 set(BIN_DIRECTORY bin)
 
+# Same soversion as OpenSSL
+set(GOST_SOVERSION "${OPENSSL_VERSION_MAJOR}.${OPENSSL_VERSION_MINOR}")
+
 set(OUTPUT_DIRECTORY ${CMAKE_SOURCE_DIR}/${BIN_DIRECTORY})
 
 #set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${OUTPUT_DIRECTORY})
@@ -110,8 +113,9 @@ add_test(NAME grasshopper
 add_library(gost STATIC ${GOST_LIB_SOURCE_FILES})
 set_target_properties(gost PROPERTIES POSITION_INDEPENDENT_CODE ON)
 
-add_library(gost_engine MODULE ${GOST_ENGINE_SOURCE_FILES})
+add_library(gost_engine SHARED ${GOST_ENGINE_SOURCE_FILES})
 set_target_properties(gost_engine PROPERTIES PREFIX "" OUTPUT_NAME "gost")
+set_target_properties(gost_engine PROPERTIES VERSION ${GOST_SOVERSION} SOVERSION ${GOST_SOVERSION})
 
 target_link_libraries(gost_engine ${OPENSSL_CRYPTO_LIBRARY} gost)
 

